app/notifications/validators.py

```diff
@@ -57,30 +57,39 @@ def check_service_over_api_rate_limit(service, key_type):
                 raise RateLimitError(rate_limit, interval, key_type)
 
 
-def check_service_over_daily_message_limit(service, key_type):
+def check_service_over_daily_message_limit(service, key_type, notification_type, num_notifications=1):
     if key_type == KEY_TYPE_TEST or not current_app.config["REDIS_ENABLED"]:
-        return 0
-
-    cache_key = daily_limit_cache_key(service.id)
-    service_stats = redis_store.get(cache_key)
-    if service_stats is None:
-        # first message of the day, set the cache to 0 and the expiry to 24 hours
-        service_stats = 0
-        redis_store.set(cache_key, service_stats, ex=86400)
-        return service_stats
-    if int(service_stats) >= service.message_limit:
-        current_app.logger.info(
-            "service {} has been rate limited for daily use sent {} limit {}".format(
-                service.id, int(service_stats), service.message_limit
+        return
+
+    rate_limits = {
+        None: service.message_limit,
```

Comment:
I flip-flopped between using `None` and `"total"` actually, so yes we could. I ended up using `None` because it's the value of `notification_type` passed in, which I think makes the lookup more obvious. I don't think there's a super clear answer, so I'll leave it as-is for now, but I'm happy to be swayed if you have strong feelings.

Reply:
No strong feelings, since you said the "total" limit will eventually go away anyway ☺️